onnx/common/tensor.h

```diff
@@ -139,6 +250,382 @@ struct Tensor final {
     has_name_ = true;
     name_ = std::move(name);
   }
+
+  bool is_raw_data() const {
+    return is_raw_data_;
+  }
+
+  //applies function f element-wise to this, storing result in this
+  //type T must correspond to appropriate tensor type
+
+  void apply_unary_function(void (*f)(float*))  {
```

Comment:
Looks to me these functions are quite similar, maybe better to use macros to reduce copy-paste like here https://github.com/onnx/onnx/blob/master/onnx/defs/schema.cc#L353-L379?

Reply:
Thought about it, but the functions may differ slightly in implementation for FLOAT16, COMPLEX64, COMPLEX128. I'm not too familiar with schema macros though so feel free to suggest if refactoring with macros is still plausible